### PR TITLE
Final goal

### PR DESCRIPTION
**Pull request by Hmekus**

### CODE REVIEW

File: RPS.html [hunk 1]
```diff
@@ -67,27 +67,61 @@
       }
     }
 
+    let playerScore = 0;
+    let computerScore = 0;
```
Comment (krabique) [inline RPS.html]: I'd rather place those in the beginning, where the `const rock` etc is. It'll be much easier to find it.

File: RPS.html [hunk 2]
```diff
@@ -67,27 +67,61 @@
...
+      const computerChoiceParagraph = document.createElement('p');
+      const playerScoreParagraph = document.createElement('p');
+      const computerScoreParagraph = document.createElement('p');
+      
+      const playerRockWin = "You won! Rock beats Scissors.";
+      const playerRockLose = "You lose! Rock beats Scissors.";
+      const playerPaperWin = "You won! Paper beats rock.";
+      const playerPaperLose = "You lose! Paper beats rock.";
+      const playerScissorsWin = "You won! Scissors beats paper.";
+      const playerScissorsLose = "You lose! Scissors beats paper";
```
Comment (krabique) [inline RPS.html]: Why did you duplicate those? Those constants must be taken from a single place throughout the script at all times.

File: RPS.html [hunk 3]
```diff
@@ -67,27 +67,61 @@
...
+
+      let roundResult = playRound(playerSelection, computerSelection);
+
+      if (roundResult === playerRockWin || roundResult === playerPaperWin || roundResult === playerScissorsWin) {
+        playerScore += 1;
+      } else if (roundResult === playerRockLose || roundResult === playerPaperLose || roundResult === playerScissorsLose) {
+        computerScore += 1;
+      }
```
Comment (krabique) [inline RPS.html]: Ok, let's refactor this. 
1. For the first if condition, we can use array's `some` in this case. https://learn.javascript.ru/array-iteration#every-some
- put winning strings in an array, like `playerRockWin`
- use `some` on the array to find out if any of the values in it that array equals `roundResult`

2. After the first if condition, we only need to check for draw, there is no need to check everything all over again. So replace the second if condition with a draw check.

File: RPS.html [hunk 4]
```diff
@@ -67,27 +67,61 @@
...
 
-      let textResultGame = [pRoundRPS, pUserSelect, pCompSelect];
+      let textResultGame = [roundResultParagraph, userChoiceParagraph, computerChoiceParagraph, playerScoreParagraph, computerScoreParagraph];
 
       textResultGame.forEach((element) => resultGameDiv.appendChild(element));
+
+      comparisonScores();
+    }
+
+    function comparisonScores() {
```
Comment (krabique) [inline RPS.html]: compareScores

File: RPS.html [hunk 5]
```diff
@@ -67,27 +67,61 @@
...
+      if (playerScore === 5 && computerScore < 5) {
+        alert("Greetings! You are a winner!");
```
Comment (krabique) [inline RPS.html]: Did you mean "congrats"? :D
Greetings means hi

File: RPS.html [hunk 6]
```diff
@@ -67,27 +67,61 @@
...
+        return; 
+      } else if (computerScore === 5 && playerScore < 5) {
+        alert("Shame, you are a loser...");
+        return;
+      }
     }
```
Comment (krabique) [inline RPS.html]: First of all, what do you need the `return`s for?
Second, the conditions on if else if are weird. You don't need what comes after && in the first if. And you don't need the second if at all, just else.

**Review by krabique — CHANGES_REQUESTED** [review]

File: RPS.html (hunk 6 above)
Comment (Hmekus) [reply RPS.html]: I think, return stops function execution

Comment (krabique) [reply RPS.html]: well, it does, but there is no code left in the function after that `if`, so it's redundant here.

**Review by krabique — APPROVED** [review]

**Pull request merged**